### ISSUE TITLE
[bug] `hooks-extra/prefer-use-state-lazy-initialization` false positive on `useState(use(promise))`

### ISSUE THREAD

**Issue by Rel1cx** [Rel1cx/eslint-react#999]
### Describe the bug

```
const [state, setState] = useState(use(promise));
```

### Expected behavior

No error.

### Platform and versions

```Text
React v19.0.0
@eslint-react/eslint-plugin v1.36.2
```

### Stack trace

```Text

```
